azkaban-common/src/main/java/azkaban/trigger/JdbcTriggerLoader.java

```diff
@@ -168,28 +168,25 @@ private synchronized void addTrigger(Connection connection, Trigger t,
       EncodingType encType) throws TriggerLoaderException {
 
     QueryRunner runner = new QueryRunner();
-
-    long id;
-
     try {
-      runner.update(connection, ADD_TRIGGER, DateTime.now().getMillis());
-      connection.commit();
-      id =
-          runner.query(connection, LastInsertID.LAST_INSERT_ID,
-              new LastInsertID());
-
-      if (id == -1L) {
-        logger.error("trigger id is not properly created.");
-        throw new TriggerLoaderException("trigger id is not properly created.");
+      FetchLastIdfromTable fetchLastId = new FetchLastIdfromTable("trigger_id", "triggers");
+      long lastTriggerId = runner.query(connection, fetchLastId.getSelectString(),
+              fetchLastId);
+      if (lastTriggerId < 0L) {
+        logger.error("Last trigger id is not properly fetched.");
+        throw new TriggerLoaderException("Last trigger id is not properly fetched.");
       }
 
-      t.setTriggerId((int) id);
+      int insertTriggerId = (int) lastTriggerId + 1;
+      runner.update(connection, ADD_TRIGGER, insertTriggerId, DateTime.now().getMillis());
+      connection.commit();
+
+      t.setTriggerId(insertTriggerId);
```

Comment:
looks like the logic here is same as what's in addExecutor, can we consolidate the duplicate logic into one method?